src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -812,8 +827,185 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
+
+            if len(packages_removed_from_installation) > 0:
+                self.composite_logger.log_debug("Recompose truncated status file json payload: {0}".format(Constants.INSTALLATION))
+                truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file=truncated_status_file, truncated_package_list=packages_retained_in_installation,
+                    substatus_message=self.__installation_substatus_msg_copy, substatus_index=installation_substatus_index)
+
+            status_file_size_in_bytes = self.__calc_status_size_on_disk(json.dumps(truncated_status_file))
+            status_file_agent_size_diff = status_file_size_in_bytes - Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES
+            size_of_max_packages_allowed_in_status -= status_file_agent_size_diff   # Reduce the max packages byte size by tombstone, new error, and escape chars byte size
+
+        self.composite_logger.log_debug("End package list truncation")
+
+        return truncated_status_file
+
+    def __split_assessment_list(self, assessment_packages):
+        """ Split assessment list into 2 parts. MIN_ASSESSMENT_PACKAGE_TO_RETAIN count as one part and remaining as the other """
+        min_packages_count = Constants.StatusTruncationConfig.MIN_ASSESSMENT_PACKAGE_TO_RETAIN
+        min_assessment_patches_to_retain, remaining_assessment_patches = (assessment_packages[:min_packages_count], assessment_packages[min_packages_count:]) \
+            if len(assessment_packages) > min_packages_count else (assessment_packages, [])
+
+        return min_assessment_patches_to_retain, remaining_assessment_patches
+
+    def __apply_truncation_process(self, assessment_packages, installation_packages, max_package_list_capacity, low_pri_index=None):
+        """ Function truncates packages from assessment and installation substatus's while always retaining a required minimum count of assessment packages """
+        installation_low_pri = []
+        installation_high_pri = installation_packages
+        # Cut assessment list into [:5], [5:]
+        min_assessment_patches_to_retain, remaining_assessment_patches = self.__split_assessment_list(assessment_packages)
+
+        if len(min_assessment_patches_to_retain) > 0:
+            max_package_list_capacity = max_package_list_capacity - self.__calc_package_payload_size_on_disk(min_assessment_patches_to_retain)
+
+        # Apply high priority (Failed, Installed) and low priority (Pending, Excluded, Not_Selected) installation logic, and keep min 5 assessment packages
+        if low_pri_index:
+            installation_high_pri = installation_packages[:low_pri_index]
+            installation_low_pri = installation_packages[low_pri_index:]
+
+        packages_retained_in_install_high_pri, packages_removed_from_inst_high_pri, remain_capacity_for_truncation = self.__apply_truncation(installation_high_pri, max_package_list_capacity)
```

Comment:
Maintain consistency between variable names: packages_retained_in_install_high_pri, packages_removed_from_inst**all**_high_pri

Reply:
Comment applies for assessment variables as well

Reply:
Comment applies for install_low_pri variables as well

Reply:
done